### ISSUE TITLE
Error no videoWithAudioFormat

### ISSUE THREAD

**Issue by ronanPoinsignon** [sealedtx/java-youtube-downloader#60]
Hello

One of my friends found a new issue.
On some videos, it is not possible to retrieve information about the videoWithAudioFormat. I give you the code here :

YoutubeDownloader downloader = new YoutubeDownloader();
		downloader.addCipherFunctionPattern(2, "\\b([a-zA-Z0-9$]{2})\\s*=\\s*function\\(\\s*a\\s*\\)\\s*\\{\\s*a\\s*=\\s*a\\.split\\(\\s*\"\"\\s*\\)");
		downloader.setParserRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.121 Safari/537.36");
		downloader.setParserRetryOnFailure(1);

		YoutubeVideo video = downloader.getVideo(videoId);

		List<AudioVideoFormat> videoWithAudioFormats = video.videoWithAudioFormats();
		System.out.println("--------------------------------");
		for(AudioVideoFormat audio : videoWithAudioFormats)
			System.out.println(audio);
		System.out.println("--------------------------------");

------------------------------------------------

output : 

--------------------------------
--------------------------------

My friend gave me three links which do not work : 
https://www.youtube.com/watch?v=wBmWvvTV0P4&list=PLt4epqdyAHQFzxW6Za-0Z90RCC57VWDO5&index=11
https://www.youtube.com/watch?v=CH27pelftXk&list=PLt4epqdyAHQFzxW6Za-0Z90RCC57VWDO5&index=7
https://www.youtube.com/watch?v=Cp56JdkmE9s&list=PLt4epqdyAHQFzxW6Za-0Z90RCC57VWDO5&index=5

And i have this thing when i am using the following function i have a lot of "error parsing format" (shown in the file) :

public VideoYtb(String url) throws YoutubeException, IOException {
		YoutubeDownloader downloader = new YoutubeDownloader();
		downloader.addCipherFunctionPattern(2, "\\b([a-zA-Z0-9$]{2})\\s*=\\s*function\\(\\s*a\\s*\\)\\s*\\{\\s*a\\s*=\\s*a\\.split\\(\\s*\"\"\\s*\\)");
		downloader.setParserRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.121 Safari/537.36");
		downloader.setParserRetryOnFailure(1);
		this.lien = url;
		this.id = Utils.getvideoId(url);
		this.titre = downloader.getVideo(id).details().title();
	}

[message (5).txt](https://github.com/sealedtx/java-youtube-downloader/files/5444891/message.5.txt)

Is the problem with me ?

**Comment by sealedtx** [comment]
@ronanPoinsignon Hi! First of all, try without these lines: 
```
downloader.addCipherFunctionPattern(2, "\b([a-zA-Z0-9$]{2})\s*=\sfunction\(\sa\s*\)\s*\{\sa\s=\sa\.split\(\s""\s*\)");
downloader.setParserRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.121 Safari/537.36")
```
I just checked your video links, everything seems to work fine. If error still exists, please, attach your error message

**Comment by ronanPoinsignon** [comment]
Hello again,

I've removed the lines but i always can't download the videos.

Here is how i am downloading the videos : 

public File downloadFromId(File folder, String videoId, String title, boolean goodVideo) throws YoutubeException, IOException, NoVideoFoundException {
		YoutubeDownloader downloader = new YoutubeDownloader();
		//downloader.addCipherFunctionPattern(2, "\\b([a-zA-Z0-9$]{2})\\s*=\\s*function\\(\\s*a\\s*\\)\\s*\\{\\s*a\\s*=\\s*a\\.split\\(\\s*\"\"\\s*\\)");
		//downloader.setParserRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.121 Safari/537.36");
		downloader.setParserRetryOnFailure(1);

		YoutubeVideo video = downloader.getVideo(videoId);

		List<AudioVideoFormat> videoWithAudioFormats = video.videoWithAudioFormats();
		Format format = null;
		if(goodVideo)
			format = getMaxVideoQuality(videoWithAudioFormats); //return the better video quality -> throw NoVideoFoundException if no video found
		else
			format = getMinVideoQuality(videoWithAudioFormats); //return the worst video quality -> throw NoVideoFoundException if no video found
		URL website = new URL(format.url());
		String titre = title;
		if(titre == null)
			titre = video.details().title();
		if(!folder.exists())
			folder.mkdirs();
		titre = Utils.removeIllegalChars(titre);
		File fichierVideo = new File(folder.getPath() + "\\" + titre + ".mp4");
		try (InputStream in = website.openStream()) {
		    Files.copy(in, fichierVideo.toPath(), StandardCopyOption.REPLACE_EXISTING);
		}
		return fichierVideo;
	}

------------------------------

I've succeeded once but only once, the other times i have a NoVideoFoundException (which is an exception i made) so i don't really understand.

**Comment by sealedtx** [comment]
@ronanPoinsignon 
1) Please, check out latest version with bug fixing: 2.4.2.
2) Attach erros and output provided by your code snippet
3) Check if there are no problem with connection to internet and to youtube (mb you are using proxy which is blocked by youtube)

**Comment by ronanPoinsignon** [comment]
Oh there is a new version i didn't see ! I think the problem was the copyright so this is fixed with the 2.4.2. Sry !
